config/config.go

```diff
@@ -1147,6 +1168,19 @@ func migrateConfig(v *viper.Viper) {
 	}
 }
 
+func migrateConfigGoRuntime(v *viper.Viper) {
```

Comment:
Looking at these migration functions, and the fact they define the old and new fields as variables at the top, I wonder if a generic migration function that is passed the old and new fields could replace many of them. Probably out of scope for this PR.